### PR TITLE
Compare unit tests for edge cases

### LINKED ISSUE #1616: Add unit tests for `types.Compare`
Let's add basic unit tests for edge cases. They should not conflict or replace integration tests.

DoD:
- there is a unit test with a couple of test cases

Note: fixing bugs (if we find new bugs) is not in the scope of the task - we can skip such test cases.

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #1616.

## Readiness checklist

* [ ] I added tests for new functionality or bugfixes.
* [ ] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [ ] I ensured that the title is good enough for the changelog.
* [ ] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [ ] I marked all done items in this checklist.

### CODE REVIEW

File: internal/types/compare_test.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+func TestCompare(t *testing.T) {
+	t.Parallel()
+
+	for name, tc := range map[string]struct {
+		skip     string
+		a        any
+		b        any
+		expected CompareResult
+	}{
+		"UnsetCompareNullTypeFieldDocument": {
```
Comment (chilagrow) [inline internal/types/compare_test.go:35]: See https://www.mongodb.com/docs/manual/reference/bson-type-comparison-order/#non-existent-fields

File: internal/types/compare_test.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+			b:        must.NotFail(NewDocument("foo", NullType{})),
+			expected: Equal,
+		},
+		"NullTypeFieldCompareUnsetDocument": {
+			skip:     "https://github.com/FerretDB/FerretDB/issues/1023",
+			a:        must.NotFail(NewDocument("foo", NullType{})),
+			b:        must.NotFail(NewDocument()),
+			expected: Equal,
+		},
+		"ArrayCompareNumber": {
```
Comment (chilagrow) [inline internal/types/compare_test.go:47]: See https://www.mongodb.com/docs/manual/reference/bson-type-comparison-order/#arrays

**Review by rumyantseva — APPROVED** [review]
I'm glad we have more clarity with these test cases! 👍

**Review by w84thesun — APPROVED** [review]
